src/scaffoldmaker/meshtypes/meshtype_3d_bladderurethra1.py

```diff
@@ -813,68 +813,37 @@ def generateBaseMesh(cls, region, options):
         if includeUreter:
-            # Left ureter
-            nodeIdentifier += 1
-            idx1 = elementsCountAlong * elementsCountAround + elementsCountAroundUreter
-            element1 = mesh.findElementByIdentifier(idx1)
-            markerUreter1Point = markerPoints.createNode(nodeIdentifier, markerTemplateInternal)
-            cache.setNode(markerUreter1Point)
-            markerName.assignString(cache, 'ureter junction of bladder trigone')
-            markerLocation.assignMeshLocation(cache, element1, [0.0, 1.0, 1.0])
-            # Right ureter
-            nodeIdentifier += 1
-            idx2 = elementsCountAlong * elementsCountAround + 2 * elementsCountAroundUreter
-            element2 = mesh.findElementByIdentifier(idx2)
-            markerUreter2Point = markerPoints.createNode(nodeIdentifier, markerTemplateInternal)
-            cache.setNode(markerUreter2Point)
-            markerName.assignString(cache, 'ureter junction of bladder trigone')
-            markerLocation.assignMeshLocation(cache, element2, [0.0, 1.0, 1.0])
+            idx2 = elementsCountAlong * elementsCountAround + elementsCountAroundUreter
+            idx3 = elementsCountAlong * elementsCountAround + 2 * elementsCountAroundUreter
+            xi2 = [0.0, 1.0, 0.0]
+            xi3 = [0.0, 1.0, 0.0]
         else:
-            # Left ureter
-            nodeIdentifier += 1
-            idx1 = ureterElementPositionDown * elementsCountAround + ureterElementPositionAround + 1
-            element1 = mesh.findElementByIdentifier(idx1)
-            markerUreter1Point = markerPoints.createNode(nodeIdentifier, markerTemplateInternal)
-            cache.setNode(markerUreter1Point)
-            markerName.assignString(cache, 'ureter junction of bladder trigone')
-            markerLocation.assignMeshLocation(cache, element1, [ureter1Position.xi1, ureter1Position.xi2, 1.0])
-            # Right ureter
+            idx2 = ureterElementPositionDown * elementsCountAround * elementsCountThroughWall + ureterElementPositionAround + 1
+            idx3 = ureterElementPositionDown * elementsCountAround * elementsCountThroughWall + elementsCountAround - ureterElementPositionAround
+            xi2 = [ureter1Position.xi1, ureter1Position.xi2, 0.0]
+            xi3 = [1 - ureter1Position.xi1, ureter1Position.xi2, 0.0]
+        idx4 = (elementsCountAlongBladder - 1) * elementsCountAround * elementsCountThroughWall + elementsCountAround // 2
+        xi4 = [1.0, 1.0, 0.0]
+        idx5 = (elementsCountAlongBladder - 1) * elementsCountAround * elementsCountThroughWall + 1
+        xi5 = [0.0, 1.0, 0.0]
+
+        element_id = [idx1, idx2, idx3, idx4, idx5]
+        xi = [xi1, xi2, xi3, xi4, xi5]
+        marker_dic = {"name": ["Apex of urinary bladder", " left ureter junction with bladder", "right ureter junction with bladder",
+                      "urethra junction with bladder dorsal", "urethra junction with bladder ventral"], "element_id": element_id, "xi": xi}
```

Comment:
Better, but I suggest you rearrange all the above code to get the names at the top and assign them to variables e.g.
```
apexMarkerName = "apex of urinary bladder"
leftUreterMarkerName = "left ureter junction with bladder"
rightUreterMarkerName = "right ureter junction with bladder"
dorsalUrethraMarkerName = "urethra junction with bladder dorsal"
ventralUrethraMarkerName = "urethra junction with bladder ventral"
```
Then you can go through all your code cases and add markers to the following list:
```
markers = []
markers.append({ "name" : apexMarkerName, "element" : element, "xi" : xi })
if includeUreter:
    ...
    markers.append({ "name" : leftUreterMarkerName , "element" : element, "xi" : xi })
    ...
else:
    ...
```
At the end you can do a for loop over the markers and add them all.

The benefit is that you have the text names written once, the language will catch you if you put an incorrect variable name in for the marker, and each record of a marker point keeps name, element and xi together. All helpful for maintenance.

Also note that you had some capitals and leading whitespace in the names in your array...